### ISSUE TITLE
[Bug]: Errors with Nostr Wallet Connect via NIP47

### ISSUE THREAD

**Issue by Nisaba** [btcpayserver/btcpayserver#6467]
### What is your BTCPay version?

2.0.3

### How did you deploy BTCPay Server?

Docker

### What happened?

We have set a connection string for an Alby wallet like this :
nostr+walletconnect://05d719xxxxxxxxxxxxxxxxxxxxxxxxx?relay=wss://relay.getalby.com/v1&secret=xxxxxxxxxxxxxxxxxxxxx&lud16=someuser@getalby.com
(note that type=nwc; is missing).
This connexion string is accepted by BTCPay and the Lightning wallet is ON:

 
![image](https://github.com/user-attachments/assets/f5de0c74-dcad-4413-9efb-2f9260f8e41c)

But payment to invoices are not detected by BTCPay. See error in log below : "System.FormatException: The key 'type' is mandatory"

But If we add  type=nwc; in the connection string, BTCPay doesn't accept:

![image](https://github.com/user-attachments/assets/ebdbc440-1a72-4590-88e7-aa52224fe274)

### How did you encounter this bug?

-

### Relevant log output

```shell
Message: The key 'type' is mandatory
Type: FormatException

Stack Trace
System.FormatException: The key 'type' is mandatory
   at BTCPayServer.Lightning.LightningConnectionStringHelper.ExtractValues(String connectionString, String& type)
   at BTCPayServer.Payments.Lightning.LightningInstanceListener.Listen(CancellationToken cancellation) in /source/BTCPayServer/Payments/Lightning/LightningListener.cs:line 511
```

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by dennisreimann** [comment]
The connection string has to be in either of the two formats, also see the details on the Lightning connection page:

- `type=nwc;key=b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&secret=71a8c14c1407c113601079c4302dab36460f0ccd0ad506f1f2dc73b5100e4f3c`
- `nostr+walletconnect:b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&secret=71a8c14c1407c113601079c4302dab36460f0ccd0ad506f1f2dc73b5100e4f3c`

I think the one you used previously is good ­— however, the invoices don't get marked as settled for me as well with the versions 1.1.15.0 and 1.1.16.0 of the Nostr plugin. 

I'll assign @Kukks for fixing this.

**Comment by Nisaba** [comment]
> But what actually works isn't documented.
> ` type=nwc;key=nostr+wallet...`

This connection string works but the invoice is still not settled after payment

**Comment by dennisreimann** [comment]
We'll release a new version tomorrow, which will fix this.

**Comment by decentralizedb** [comment]
I'm reusing this issue but I don't know if it would be better to open a different issue.
Yesterday I was having the same problem describing this issue.

I'm trying to connect a store to a LN node, CLN. 
I'm using NWC. I'm generating the connection with the CLN-NWC plugin from CLN.
When I set the string to validate the connection in the BTCPay, I get the following message:
"The network of the wallet (bitcoin) does not match the network of the server (Mainnet)"

BTCPay is an advanced installation, outside of docker or Lunanode (all components installed manually).
I have tested the connection using GetAlby, and it works without any problem.
On logs, I can't see any message related to this error.

Any idea?

BTCPayserver: v2.0.4
CLN: v24.11
cln_nwc: [v0.1.2](https://github.com/gudnuf/cln_nwc/releases/tag/v0.1.2)
SO: Ubuntu 20.04

![image](https://github.com/user-attachments/assets/2ee329d2-181f-4a6f-bb86-829a94f2d932)
